service/gcs/core/gcs/gcs.go

```diff
@@ -47,14 +48,71 @@ type gcsCore struct {
 	// into the gcsCore. It is structured as a map from pid to cache entry.
 	processCache map[int]*processCacheEntry
 
+	// baseLogPath is the path where all container logs should be nested.
+	baseLogPath string
+
 	// baseStoragePath is the path where all container storage should be nested.
 	baseStoragePath string
+
+	// containerIndexMutex is to lock access to the containerIndex slice
+	containerIndexMutex sync.Mutex
+
+	// containerIndex is a slice that tracks the index that the container was
+	// created in based on its array location and is aggressively reused when a
+	// location is no longer needed.
+	containerIndex []string
+}
+
+// getOrAddContainerIndex gets the index that a container was already inserted
+// at or inserts the container at the next available index. This method will
+// panic of 255+ containers.
+func (c *gcsCore) getOrAddContainerIndex(id string) uint8 {
+	c.containerIndexMutex.Lock()
+	defer c.containerIndexMutex.Unlock()
+
+	len := len(c.containerIndex)
+	insertAt := len
+	for i := 0; i < len; i++ {
+		if c.containerIndex[i] == "" && i <= insertAt {
+			insertAt = i
+		} else if c.containerIndex[i] == id {
+			// We already have inserted this id. Return its index.
+			return uint8(i)
+		}
+	}
+
+	if insertAt > math.MaxUint8 {
+		panic("Maximum number of container indexes hit")
```

Comment:
Would we want to panic here? What if a user just adds 256 containers? Wouldn't we want it to just be an error instead?

Reply:
Yea I can do error.